autogpt_platform/backend/backend/integrations/webhooks/stripe.py

```
@@ -0,0 +1,183 @@
+
+    async def _register_webhook(
+        self,
+        credentials: Credentials,
+        webhook_type: StripeWebhookType,
+        resource: str,
+        events: list[str],
+        ingress_url: str,
+        secret: str,
+    ) -> tuple[str, dict]:
+        if not isinstance(credentials, APIKeyCredentials):
+            raise ValueError("Stripe webhook registration requires an API key")
```

Ruff (0.16.1)

[warning] Prefer `TypeError` exception for invalid type

(TRY004)